src/codegen/utilities/enum_helpers.py

```diff
@@ -0,0 +1,283 @@
+    'M_VOLTS': 'MILLIVOLTS',
+    'ON_BRD': 'ONBRD',
+    'US_BBULK': 'USB_BULK',
+    '0TO_1_V': 'ZERO_TO_ONE_V',
+    '0TO_5_V': 'ZERO_TO_FIVE_V',
+    '0TO_10_V': 'ZERO_TO_TEN_V',
+    '0TO_20M_A': 'ZERO_TO_TWENTY_M_A',
+    'NEG_1ZERO_TO_TEN_V': 'NEG_10_TO_10_V',
+    '10_MHZ_REF_CLOCK': 'TEN_MHZ_REF_CLOCK',
+
+}
+
+# Special case for Impedance1 where the enum value name is numeric
+IMPEDANCE_ENUM_VALUE_NAME_SUB = {
+    '50': 'FIFTY_OHMS',
+    '75': 'SEVENTY_FIVE_OHMS',
+    '1000000': 'ONE_M_OHM',
+    '10000000000': 'TEN_G_OHMS'
+}
+
+
+# TODO: bitfield types
+
+def _merge_enum_values(valueses):
+    result_set = {}
+    for values_array in valueses:
+        for value in values_array:
+            value_num = value['value']
+            # If it exists already, only overwrite if the current one has no documentation.
+            if value_num not in result_set or 'documentation' not in result_set[value_num]:
+                result_set[value_num] = value
+
+    return list(result_set.values())
+
+
+def _merge_enum_variants(enums):
+    # Combine the numbered enum variants. These exist to give remove options that aren't applicable
+    # for some attributes in interactive environments like G Controls/Indicators and CVI Function
+    # Panels.
+    name_pattern = re.compile("(.*\D)(\d+)")
+
+    enum_merge_set = {}
+
+    for _, enum_name in enumerate(sorted(enums.keys())):
+        match = name_pattern.fullmatch(enum_name)
+        if match:
+            basename = match.group(1)
+
+            if basename in ENUMS_BLACKLIST and enum_name in enums.keys():
+                enums.pop(enum_name) 
+
+            if basename in enums.keys():
+                if basename not in enum_merge_set.keys():
+                    enum_merge_set[basename] = []
+                    enum_merge_set[basename].append(basename)  
+                if  enum_name not in enum_merge_set[basename]:
+                    enum_merge_set[basename].append(enum_name)    
+
+
+    for basename, enums_to_merge in enum_merge_set.items():
+        _logger.debug(f"merging enums: {basename} <-- {enums_to_merge}")
+        enums[basename] = {
+            'values': _merge_enum_values([enums[enum]['values'] for enum in enums_to_merge])
+        }
+        # delete the variants, now
+        for enum in enums_to_merge:
+            if not enum == basename:
+                del enums[enum]
+
+    # sort it by key (enum name)
+    return dict(sorted(enums.items()))
+
+
+def _sanitize_values(enums):
+    for _, enum in enums.items():
+        for value in enum['values']:
+            value_name = value['name']
+    
+            # For Impedance1
+            if value_name in IMPEDANCE_ENUM_VALUE_NAME_SUB.keys():
+                value_name = IMPEDANCE_ENUM_VALUE_NAME_SUB[value_name]
+                
+            for old, new in NAME_SUBSTITUTIONS.items():
+                value_name = value_name.replace(old, new)
```

Comment:
When we do enum substitutions here, we don't do the same substitution in docstrings that reference the enum. (Maybe we could, but we would have to be careful about matching substrings.)

When we do enum substitutions in scrapigen, they cover both the enum and any docstrings that reference the enum.

Reply:
We have handled the docstrings using this function and look like we don't have any case where enum name and enum docstring have the same substitutions.https://github.com/ni/nidaqmx-python/blob/9e1b171a01bf9fd7bd057a55ec70eb620ce9548e/src/codegen/utilities/enum_helpers.py#L165